src/databricks/labs/lsql/dashboards.py

```diff
@@ -204,16 +204,33 @@ def create_dashboard(self, dashboard_folder: Path) -> Dashboard:
         lakeview_dashboard = Dashboard(datasets=datasets, pages=[page])
         return lakeview_dashboard
 
-    @staticmethod
-    def _get_datasets(dashboard_folder: Path) -> list[Dataset]:
+    def _get_datasets(self, dashboard_folder: Path, *, database: str = "") -> list[Dataset]:
         datasets = []
         for query_path in sorted(dashboard_folder.glob("*.sql")):
-            with query_path.open("r") as query_file:
-                raw_query = query_file.read()
-            dataset = Dataset(name=query_path.stem, display_name=query_path.stem, query=raw_query)
+            query = self._replace_database_in_query(query_path.read_text(), database)
+            dataset = Dataset(name=query_path.stem, display_name=query_path.stem, query=query)
             datasets.append(dataset)
         return datasets
 
+    @staticmethod
+    def _replace_database_in_query(query: str, database: str) -> str:
+        if not database:
+            return query
+
+        try:
+            syntax_tree = sqlglot.parse_one(query, dialect=sqlglot.dialects.Databricks)
+        except sqlglot.ParseError as e:
+            logger.warning(f"Parsing {query}: {e}")
+            return query
+
+        def transformer(node: sqlglot.Expression) -> sqlglot.Expression:
```

Comment:
@nfx : maybe we want the users to able to inject the transformer.

This flexibility would come in hand when limiting the size in development by appending a `LIMIT 1000` to the query. Or, for example with dbt, it is common to have multiple databases/schemas in development following a certain convention (usually a prefix but can be overwritten):

``` SQL
SELECT sales.a, campaigns.b FROM cor_financials.sales AS sales JOIN cor_marketing.campaigns AS campaigns
```

In this case we would like to prefix the databases with `cor_` instead of a replace:

> Note This is SQL after `dbt compile`, the `cor_financials.sales` was something like `{{ ref("mart_sales") }}` before compiling. 

And, for the cli, we can force it to only do database replacement.

Reply:
If we allow for injection of the transformer, we need to make sure that the `find_fields` logic is using the transformed SQL

Reply:
The logic is rewritten to support injecting a sqlglot transformer. There is a transformer function to replace the database, the test show how to